subsys/bluetooth/host/conn.c

```diff
@@ -926,7 +926,7 @@ void bt_conn_process_tx(struct bt_conn *conn)
 	err = send_buf(conn, buf);
 	net_buf_unref(buf);
 
-	if (err  == -EIO) {
+	if (err == -EIO || err == -ENOTCONN) {
```

Comment:
Isn't this already being done in 
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/host/conn.c#L643-L656

Reply:
Yeah that's the intention at least. But this PR proves that the flow is obfuscated enough that no one here is really sure of what's happening anymore. 
I had a branch that massively simplifies all of this, but couldn't merge it as it exposed other issues. I'll see if we can split out some parts of it.

@watsug I took the liberty of editing your PR message to link the issue you're fixing, in order to give more context to the reviewers .

Reply:
@jori-nordic What should we do with this PR? 

It may sound like it's fixing an issue, but it also sounds like that the fix here already exist and has just been moved, so to me it doesn't seem to actually change anything.

Reply:
I'll have another look at the explanation (next week) and report back.

Sometimes, fixes like these are impossible to reproduce in bsim upstream, as the ZLL is not compatible with `BT_RECV_WORKQ_SYS` and `BT_RECV_WORKQ_BT`. These two options can lead to a very different scheduling pattern, highlighting issues like this one.

Reply:
Instead of singling out "problematic" error codes, we could only check for the cases where we want to keep the buffer in the queue, E.g. -EAGAIN.
TODO for myself: document what error codes should trigger a destroy.